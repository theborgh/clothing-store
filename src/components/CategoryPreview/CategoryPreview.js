import React from 'react';
import ShopItem from '../ShopItem/ShopItem';
import './CategoryPreview.scss';

const categoryPreview = ({ title, items }) => {
   const PREVIEW_ITEMS_COUNT = 4;

   return (
      <div className="CategoryPreview">
         <h1 className="title">{title.toUpperCase()}</h1>
         <div className="itemPreview">
            {
               items
                  .filter((item, i) => i < PREVIEW_ITEMS_COUNT)
                  .map((item) => (
                     <ShopItem key={item.id} item={item} />
                  ))
            }
         </div>
      </div>
   );
}

export default categoryPreview;