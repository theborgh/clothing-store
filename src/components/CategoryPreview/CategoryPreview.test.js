import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import CategoryPreview from './CategoryPreview';

Enzyme.configure({ adapter: new Adapter() });

const mockItems = [
  {
      id: 345,
  }
];

describe("CategoryPreview", () => {
  it("Renders 1 component", () => {
    expect(Enzyme.shallow(<CategoryPreview title="Heh" items={mockItems} />).length).toEqual(1);
  });
  it("Matches previous snapshot", () => {
    expect(Enzyme.shallow(<CategoryPreview title="Heh" items={mockItems} />).debug()).toMatchSnapshot();
  });
});