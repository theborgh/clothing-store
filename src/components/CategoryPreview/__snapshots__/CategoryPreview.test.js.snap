// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryPreview Matches previous snapshot 1`] = `
"<div className=\\"CategoryPreview\\">
  <h1 className=\\"title\\">
    HEH
  </h1>
  <div className=\\"itemPreview\\">
    <Connect(shopItem) item={{...}} />
  </div>
</div>"
`;
